### PR TITLE
Better handling of sensors

### PR DESCRIPTION
**Pull request by DrewBrandt**
This one has some other changes as well, but is still relatively minor in scope. It should be tested before merging.
It should also be merged after the other pull request I wrote, #72.

Changes:
- Allows `if(Sensor)` check to see if sensor was initialized. Same as doing `sensor.isInitialized()`
- Neatened state init to bring all flags into constructor instead of init.
- Added `getHasFirstFix()` to gps
- Changed behavior of state sensor storage to support multiple sensors of the same type, found by `(getSensor(SensorType type, optional int sensorNum)` (1 indexed) so the first baro would be `getSensor(BAROMETER_)` and the second barometer would be `getSensor(BAROMETER_, 2)`. The order of sensors is defined by the user when they use `addSensor(SensorType type, optional int sensorNum)` (1 indexed). This will overwrite the first sensor if no number is given, rather than adding on to the end.
- Changed launch detection conditions to verify that at least one of the barometer or IMU is working. Otherwise, it will not detect launch and simply continue printing pre launch to the SD card.

### CODE REVIEW

**Review by varun-un — COMMENTED** [review]
Test pls

File: Spaceport/23-24/Code/Teensy-Based Avionics/src/MAX_M10S.h [hunk 1]
```diff
@@ -28,23 +28,22 @@ class MAX_M10S : public GPS
...
+    int getFixQual() override;
+    double getHeading() override;
 
     bool initialize() override;
     const char *getCsvHeader() override;
     char *getDataString() override;
     char *getStaticDataString() override;
     const char *getName() override;
     void update() override;
+    bool getHasFirstFix() override;
```
Comment (varun-un) [inline Spaceport/23-24/Code/Teensy-Based Avionics/src/MAX_M10S.h:46]: name's a little goofy but it's fine

File: Spaceport/23-24/Code/Teensy-Based Avionics/src/main.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@ DS3231 rtc();               // I2C Address 0x68
 APRSConfig config = {"KC3UTM", "APRS", "WIDE1-1", '[', '/'};
 RadioSettings settings = {433.775, true, false, &hardware_spi, 10, 31, 32};
 RFM69HCW radio = {settings, config};
-State computer;
+State computer;// = useKalmanFilter = true, stateRecordsOwnData = true
```
Comment (varun-un) [inline Spaceport/23-24/Code/Teensy-Based Avionics/src/main.cpp:17]: what are these settings for `useKalmanFilter`, is it just a random comment or is this like you have to configure this somewhere?

**Review by varun-un — APPROVED** [review]
Looks good, you can ignore the two commits I made on top, they're irrelevant

File: Spaceport/23-24/Code/Teensy-Based Avionics/src/main.cpp (hunk 2 above)
Comment (DrewBrandt) [reply Spaceport/23-24/Code/Teensy-Based Avionics/src/main.cpp:17]: It's the default values for the constructor. you can set useKalmanFilter to false or stateRecordsownData to false, but by default they are true.

**Pull request merged**